crates/vm/vm.rs

```diff
@@ -103,56 +159,49 @@ cfg_if::cfg_if! {
                     }
                 }
             }
-            let mut receipts = Vec::new();
-            let mut cumulative_gas_used = 0;
 
             let store_wrapper = Arc::new(StoreWrapper {
                 store: state.database().unwrap().clone(),
                 block_hash: block.header.parent_hash,
             });
 
-            let mut account_updates: Vec<AccountUpdate> = vec![];
+            // Account updates are initialized like this because of the beacon_root_contract_call, it is going to be empty if it wasn't called.
+            let mut account_updates = get_state_transitions(state);
 
-            for transaction in block.body.transactions.iter() {
-                let result = execute_tx_levm(transaction, block_header, store_wrapper.clone()).unwrap();
-                cumulative_gas_used += result.gas_used;
-                let receipt = Receipt::new(
-                    transaction.tx_type(),
-                    matches!(result.result, TxResult::Success),
-                    cumulative_gas_used,
-                    result.logs,
-                );
-                receipts.push(receipt);
+            let mut receipts = Vec::new();
+            let mut cumulative_gas_used = 0;
+            let mut temporary_cache: CacheDB = HashMap::new();
```

Comment:
I think `block_cache` or `block_execution_cache` are proper names for this as this.

Reply:
Great idea! Done [here](https://github.com/lambdaclass/ethrex/pull/1546/commits/30eecfe3384719a639d5d610816b35737c098676)